plugins/qeta/src/components/MarkdownEditor/MarkdownEditor.tsx

```diff
@@ -1,23 +1,104 @@
+/* eslint-disable no-console */
 import React from 'react';
 import ReactMde from 'react-mde';
+
+import { Config } from '@backstage/config';
 import { MarkdownContent } from '@backstage/core-components';
 import 'react-mde/lib/styles/css/react-mde.css';
 import 'react-mde/lib/styles/css/react-mde-editor.css';
 import 'react-mde/lib/styles/css/react-mde-toolbar.css';
 import { useStyles } from '../../utils/hooks';
+import FileType from 'file-type';
+import { ErrorApi, errorApiRef, useApi } from '@backstage/core-plugin-api';
+
+const DEFAULT_IMAGE_SIZE_LIMIT = 2500000;
+const SUPPORTED_FILES_TYPES = [
+  'image/png',
+  'image/jpg',
+  'image/jpeg',
+  'image/gif',
+];
+
+export const submitImage = async (
+  config: Config,
+  file: Blob,
+  erroAlert: ErrorApi,
+) => {
+  const backendBaseUrl = config.getString('backend.baseUrl');
+  const qetaUrl = `${backendBaseUrl}/api/qeta/attachments`;
+  const formData = new FormData();
+
+  formData.append('image', file);
+
+  const requestOptions = {
+    method: 'POST',
+    body: formData,
+  };
+
+  const response = await fetch(qetaUrl, requestOptions);
+
+  if (response.status >= 400) {
+    const responseError = await response.text();
+    const error = new Error(
+      `Failed to upload image question : ${responseError}`,
+    );
+    erroAlert.post(error);
+    throw new Error(`Failed to upload image question : ${responseError}`);
+  }
+  return response.json();
+};
+
+const imageUpload = (config: Config, erroAlert: ErrorApi) => {
+  const maxSizeImage =
+    config?.getOptionalNumber('qeta.storage.maxSizeImage') ||
+    DEFAULT_IMAGE_SIZE_LIMIT;
+
+  // eslint-disable-next-line func-names
+  return async function* (data: ArrayBuffer) {
+    const fileType = await FileType.fromBuffer(data);
+
+    if (fileType && !SUPPORTED_FILES_TYPES.includes(fileType?.mime)) {
+      erroAlert.post(new Error(`Image type (${fileType.mime}) not supported.`));
+      return false;
+    }
+
+    if (data.byteLength > maxSizeImage) {
+      erroAlert.post(
+        new Error(
+          `Image larger than ${maxSizeImage} bytes try to make it smaller before uploading.`,
+        ),
+      );
+      return false;
+    }
+    const { locationUri } = await submitImage(
+      config,
+      new Blob([data], { type: 'text/plain' }),
```

Comment:
Why text/plain even you know the type here already?

Reply:
I probably ended up forgetting after the refactor I'll update 😅